src/components/actions/processTrigger.ts

```diff
@@ -82,20 +87,38 @@ export async function processTrigger(payload: TriggerPayload) {
         parameter_queries: action.parameter_queries,
         trigger_payload: action.trigger_payload,
       }
-      const result = await executeAction(actionPayload, actionLibrary, {
-        outputCumulative,
-      })
+      const result = await executeAction(
+        actionPayload,
+        actionLibrary,
+        {
+          outputCumulative,
+        },
+        previewData
+      )
+
       outputCumulative = { ...outputCumulative, ...result.output }
+      if (result.status !== ActionQueueStatus.ConditionNotMet)
```

Comment:
Hmm, i think this should really be filtered out at the consumption end, it's still actionOutput

Reply:
Also I've noticed the FAIL ones are coming through to displayData return

Reply:
> Hmm, i think this should really be filtered out at the consumption end, it's still actionOutput

No, cos those actions never run, so they don't have any valid or useful output. I don't see the point in this info being returned to the front end. For example, when previewing a decision, there's three different actions, one for each of APPROVED, REJECTED and LOQ. 2 out of 3 never run as they don't meet the condition, so the front-end preview should only be interested in the one that *did* run.

> Also I've noticed the FAIL ones are coming through to displayData return

Yeah, that was intentional. The front-end displays an error message for the failed ones.